src/plugins/intel_npu/src/plugin/npuw/llm_infer_request.cpp

```diff
@@ -107,8 +108,33 @@ void copy_columns_by_row_chunks(ov::SoPtr<ov::ITensor> src, ov::SoPtr<ov::ITenso
     }
 }
 
-void copy_tensor_by_dim(ov::SoPtr<ov::ITensor> src_tensor, ov::SoPtr<ov::ITensor> dst_tensor, uint32_t kv_dim) {
-    if (kv_dim == 3u) {
+void copy_tensor_by_dim(ov::SoPtr<ov::ITensor> src_tensor,
+                        ov::SoPtr<ov::ITensor> dst_tensor,
+                        uint32_t kv_dim_src,
+                        uint32_t kv_dim_dst) {
+    if (kv_dim_src != kv_dim_dst) {
+        // new case - do a generic copy for now (in fact it is a permute)
```

Comment:
I think we may leave it as is for now, but isn't first dimension is a batch and isn't it always == 1?

In this case we don't need 4d permute. And we have already covered and optimized most (but not all) cases for 3d permute.

Reply:
we don't need a generic 4d permute for sure, but I didn't want to handle the particular cases here as it'd complicate the flow at this stage